File: DataAccess/Enterprise/Models/DemandTrendInput.vb
Imports System.Xml.Serialization

<Serializable>
Public Class DemandTrendInput
  <XmlAttribute>
  Public Property Grouping As String
  <XmlAttribute>
  Public Property FIKey As Integer
  <XmlAttribute>
  Public Property StartDate As Date
  <XmlAttribute>
  Public Property EndDate As Date

  Public Property DemandLocations As List(Of Integer)

  Public Sub New()
  End Sub

  Public Sub New(fiKey As Integer, startDate As Date, endDate As Date, grouping As String, demandLocations As List(Of Integer))
    Me.FIKey = fiKey
    Me.StartDate = startDate
    Me.EndDate = endDate
    Me.Grouping = grouping
    Me.DemandLocations = demandLocations
  End Sub
End Class

File: CSharpDataAccess/Enterprise/Models/DemandTrendInput.cs
using System;
using System.Collections.Generic;
using System.Xml.Serialization;

namespace CSharpDataAccess.Enterprise.Models
{
  [Serializable()]
  public class DemandTrendInput
  {
    [XmlAttribute()]
    public string Grouping { get; set; }
    [XmlAttribute()]
    public int FIKey { get; set; }
    [XmlAttribute()]
    public DateTime StartDate { get; set; }
    [XmlAttribute()]
    public DateTime EndDate { get; set; }

    public List<int> DemandLocations { get; set; }

    public DemandTrendInput()
    {
    }

    public DemandTrendInput(int fiKey, DateTime startDate, DateTime endDate, string grouping, List<int> demandLocations)
    {
      FIKey = fiKey;
      StartDate = startDate;
      EndDate = endDate;
      Grouping = grouping;
      DemandLocations = demandLocations;
    }
  }
}
